src/pg_to_evalscript/javascript_datacube/DataCube.js

```diff
@@ -443,4 +495,172 @@ class DataCube {
+                this.insertIntoDimension(axis, dataToInsert, origSize + j)
+            }
+
+            this.dimensions[axis].labels = this.dimensions[axis].labels.concat(dimension.labels)
+        }
+    }
+
+    merge(cube2, overlap_resolver) {
+        const sharedDimensions = []
+        const cube1SpecificDimensions = []
+        const cube2SpecificDimensions = []
+
+        let dimensionWithDifferentLabels;
+        let dimensionWithDifferentLabelsOverlaps = false;
+
+        for (let dimension of this.dimensions) {
+            const dimension2 = cube2.getDimensionByName(dimension.name);
+
+            if (!dimension2) {
+                cube1SpecificDimensions.push(dimension)
+                continue
+            } else {
+                sharedDimensions.push(dimension)
+            }
+
+            const labelsEqual = dimension.labels.length === dimension2.labels.length && dimension.labels.every((l) => dimension2.labels.includes(l))
+
+            if (
+                dimension.name === dimension2.name &&
+                dimension.type === dimension2.type &&
+                labelsEqual
+            ) {
+                continue;
+            }
+
+
+            if (labelsEqual) {
+                throw new ProcessError({
+                    name: "Internal",
+                    message: "Shared dimensions have to have the same name and type in 'merge_cubes'."
+                })
+            }
+
+            if (dimensionWithDifferentLabels) {
+                throw new ProcessError({
+                    name: "Internal",
+                    message: "Only one of the dimensions can have different labels in 'merge_cubes'."
+                })
+            }
+
+            dimensionWithDifferentLabels = dimension.name
+
+            if (dimension.labels.some((l) => dimension2.labels.includes(l))) {
+                dimensionWithDifferentLabelsOverlaps = true;
+            }
+        }
+
+        for (let dimension2 of cube2.dimensions) {
+            const dimension = this.getDimensionByName(dimension2.name);
+            if (!dimension) {
+                cube2SpecificDimensions.push(dimension2)
+            }
+        }
+
+        const allDimensionsEqual = cube1SpecificDimensions.length === 0 && cube2SpecificDimensions.length === 0;
+
+        if (!overlap_resolver && ((dimensionWithDifferentLabels && dimensionWithDifferentLabelsOverlaps) || (!dimensionWithDifferentLabels && allDimensionsEqual))) {
+            throw new ProcessError({
+                name: "OverlapResolverMissing",
+                message: "Overlapping data cubes, but no overlap resolver has been specified."
+            });
+        }
+
+        if (allDimensionsEqual && !dimensionWithDifferentLabels) {
+            return this._merge_matching_cube(cube2, overlap_resolver)
+        }
+
+        const isCube2Subcube = !dimensionWithDifferentLabels && cube2SpecificDimensions.length === 0 && cube1SpecificDimensions.length > 0;
```

Comment:
Can it also be the other way around and cube1 is subcube of cube2 and low dimension cube (cube1) is being merged with higher dimension cube(cube2)? Cube1 has spatial dimension with x label, cube2 has spatial dimension with x and y.

Most likely this is (also) handled by code below.

Reply:
Nice spot, it doesn't work, added a test and changes to handle the opposite order.